src/Magento/FunctionalTestingFramework/Module/MagentoWebDriver.php

```diff
@@ -636,6 +636,23 @@ public function fillSecretField($field, $value)
         $this->fillField($field, $decryptedValue);
     }
 
+    /**
+     * Function used to create data that contains sensitive credentials in a <createData> <field> override.
+     * The data is decrypted immediately prior to data creation to avoid exposure in console or log.
+     *
+     * @param string $command
+     * @throws TestFrameworkException
+     * @return void
+     */
+    public function magentoCliSecret($command)
+    {
+        // to protect any secrets from being printed to console the values are executed only at the webdriver level as a
+        // decrypted value
+
+        $decryptedCommand = CredentialStore::getInstance()->decryptSecretValue($command);
```

Comment:
So we need to create a new function in CredentialManager to scan and replace secrets. Consider the following:

* We pass in `$command = config:set some/path 1234qewr!!` where `1234qewr!!` is the encrypted value
* We need to scan the *entire* string for secrets, instead of just trying to decrypt the whole thing.

I pinged you with the suggestion function, let me know what you think

Reply:
Added the recommended function. Verified the PHP was generated correctly.